File: apps/waddle_controller/src/util/weatherLocationCategory.test.ts
import { describe, expect, it } from 'vitest';
import {
  interestCategoryLabel,
  weatherLocationCategoryFromName,
} from './weatherLocationCategory';

describe('weatherLocationCategoryFromName', () => {
  it('maps US state suffix to north_america', () => {
    expect(weatherLocationCategoryFromName('Boston, MA')).toBe('north_america');
  });

  it('maps country suffix to continental region', () => {
    expect(weatherLocationCategoryFromName('Paris, France')).toBe('europe');
    expect(weatherLocationCategoryFromName('Sydney, Australia')).toBe('oceania');
  });
});

describe('interestCategoryLabel', () => {
  const curator = [{ id: 'united_states', label: 'United States' }];

  it('uses curator label when present', () => {
    expect(interestCategoryLabel('united_states', curator)).toBe('United States');
  });

  it('title-cases unknown slug ids', () => {
    expect(interestCategoryLabel('united_kingdom', curator)).toBe('United Kingdom');
  });

  it('treats missing category as general', () => {
    expect(interestCategoryLabel(undefined, curator)).toBe('General');
    expect(interestCategoryLabel(null, curator)).toBe('General');
    expect(interestCategoryLabel('', curator)).toBe('General');
  });
});

File: packages/waddle_shared/test/persistence/weather_location_category_test.dart
import 'package:test/test.dart';
import 'package:waddle_shared/persistence/weather_location_category.dart';

void main() {
  group('weatherLocationCategoryFromName', () {
    test('maps US city with state abbreviation to north_america', () {
      expect(weatherLocationCategoryFromName('Seattle, WA'), kWeatherLocationRegionNorthAmerica);
      expect(weatherLocationCategoryFromName('Washington, DC'), kWeatherLocationRegionNorthAmerica);
    });

    test('maps trailing country name to continental region', () {
      expect(
        weatherLocationCategoryFromName('London, United Kingdom'),
        kWeatherLocationRegionEurope,
      );
      expect(weatherLocationCategoryFromName('Tokyo, Japan'), kWeatherLocationRegionAsia);
      expect(weatherLocationCategoryFromName('Toronto, Canada'), kWeatherLocationRegionNorthAmerica);
      expect(weatherLocationCategoryFromName('Sydney, Australia'), kWeatherLocationRegionOceania);
      expect(
        weatherLocationCategoryFromName('São Paulo, Brazil'),
        kWeatherLocationRegionSouthAmerica,
      );
      expect(weatherLocationCategoryFromName('Cairo, Egypt'), kWeatherLocationRegionAfrica);
    });

    test('returns general when country segment missing', () {
      expect(weatherLocationCategoryFromName('Seattle'), 'general');
    });
  });

  group('weatherLocationRegionFromCountrySlug', () {
    test('maps unknown country slug to general', () {
      expect(weatherLocationRegionFromCountrySlug('atlantis'), 'general');
    });
  });
}
